Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc

```diff
@@ -116,105 +116,107 @@ void HcalIsoTrackAnalyzer::analyze(edm::Event const& iEvent, edm::EventSetup con
 #endif
-      t_gentrackP = itr->gentrackP_;
-      t_rhoh = itr->rhoh_;
-      t_selectTk = itr->selectTk_;
-      t_qltyFlag = itr->qltyFlag_;
-      t_qltyMissFlag = itr->qltyMissFlag_;
-      t_qltyPVFlag = itr->qltyPVFlag_;
+      t_gentrackP = itr.gentrackP_;
+      t_rhoh = itr.rhoh_;
+      t_selectTk = itr.selectTk_;
+      t_qltyFlag = itr.qltyFlag_;
+      t_qltyMissFlag = itr.qltyMissFlag_;
+      t_qltyPVFlag = itr.qltyPVFlag_;
 #ifdef EDM_ML_DEBUG
       if (debug)
         edm::LogVerbatim("HcalIsoTrack") << "gentrackP " << t_gentrackP << " rhoh " << t_rhoh
                                          << " qltyFlag:qltyMissFlag:qltyPVFlag:selectTk " << t_qltyFlag << ":"
                                          << t_qltyMissFlag << ":" << t_qltyPVFlag << ":" << t_selectTk;
 #endif
-      t_trgbits = itr->trgbits_;
-      t_DetIds = itr->detIds_;
-      t_DetIds1 = itr->detIds1_;
-      t_DetIds3 = itr->detIds3_;
+      t_trgbits = itr.trgbits_;
+      t_DetIds = itr.detIds_;
+      t_DetIds1 = itr.detIds1_;
+      t_DetIds3 = itr.detIds3_;
       if (useRaw_ == 1) {
-        t_eHcal = itr->eHcalAux_;
-        t_eHcal10 = itr->eHcal10Aux_;
-        t_eHcal30 = itr->eHcal30Aux_;
-        t_HitEnergies = itr->hitEnergiesAux_;
-        t_HitEnergies1 = itr->hitEnergies1Aux_;
-        t_HitEnergies3 = itr->hitEnergies3Aux_;
+        t_eHcal = itr.eHcalAux_;
+        t_eHcal10 = itr.eHcal10Aux_;
+        t_eHcal30 = itr.eHcal30Aux_;
+        t_HitEnergies = itr.hitEnergiesAux_;
+        t_HitEnergies1 = itr.hitEnergies1Aux_;
+        t_HitEnergies3 = itr.hitEnergies3Aux_;
       } else if (useRaw_ == 2) {
-        t_eHcal = itr->eHcalRaw_;
-        t_eHcal10 = itr->eHcal10Raw_;
-        t_eHcal30 = itr->eHcal30Raw_;
-        t_HitEnergies = itr->hitEnergiesRaw_;
-        t_HitEnergies1 = itr->hitEnergies1Raw_;
-        t_HitEnergies3 = itr->hitEnergies3Raw_;
+        t_eHcal = itr.eHcalRaw_;
+        t_eHcal10 = itr.eHcal10Raw_;
+        t_eHcal30 = itr.eHcal30Raw_;
+        t_HitEnergies = itr.hitEnergiesRaw_;
+        t_HitEnergies1 = itr.hitEnergies1Raw_;
+        t_HitEnergies3 = itr.hitEnergies3Raw_;
       } else {
-        t_eHcal = itr->eHcal_;
-        t_eHcal10 = itr->eHcal10_;
-        t_eHcal30 = itr->eHcal30_;
-        t_HitEnergies = itr->hitEnergies_;
-        t_HitEnergies1 = itr->hitEnergies1_;
-        t_HitEnergies3 = itr->hitEnergies3_;
+        t_eHcal = itr.eHcal_;
+        t_eHcal10 = itr.eHcal10_;
+        t_eHcal30 = itr.eHcal30_;
+        t_HitEnergies = itr.hitEnergies_;
+        t_HitEnergies1 = itr.hitEnergies1_;
+        t_HitEnergies3 = itr.hitEnergies3_;
       }
 #ifdef EDM_ML_DEBUG
       if (debug)
         edm::LogVerbatim("HcalIsoTrack") << "eHcal:eHcal10:eHCal30 " << t_eHcal << ":" << t_eHcal10 << t_eHcal30;
 #endif
       tree->Fill();
+      edm::LogVerbatim("HcalIsoTrackX") << "Run " << t_Run << " Event " << t_Event << " p " << t_p;
```

Comment:
Shouldn't this be fenced into the `EDM_ML_DEBUG`?

Reply:
I prefer to keep it out of EDM_ML_DEBUG protection